gusto/time_discretisation.py

```diff
@@ -471,39 +411,23 @@ def rhs(self):
 
         return r.form
 
-    def solve_stage(self, x_in, stage):
-        """
-        Perform a single stage of the Runge-Kutta scheme.
-
-        Args:
-            x_in (:class:`Function`): field at the start of the stage.
-            stage (int): index of the stage.
-        """
-        if stage == 0:
-            self.solver.solve()
-            self.k1.assign(self.x_out)
-            self.x1.assign(x_in + 0.5 * self.dt * self.k1)
+    def solve_stage(self, x0, stage):
+        self.x1.assign(x0)
```

Comment:
I think we need to include the call to evaluate physics terms here (this was included after you branched off from trunk!). But you should check if makes sense to evaluate the physics sources at exactly this point...
```suggestion
        self.x1.assign(x0)
        for evaluate in self.evaluate_source:
            evaluate(self.x1, self.dt)
```

Reply:
I've added this!